### PR TITLE
Use a secret for db credentials

### LINKED ISSUE #8: Expect DB connection information in a secret as a prerequisite
We should use k8s secrets for the RelStorage database connection, even if it makes the setup a bit more complicated. Imho the most straightforward way would be to expect the database connection secret as a prerequisite, just like the database PVC.

### PR DESCRIPTION
**Pull request by tschorr**
Fix #8.

### CODE REVIEW

**Review by xiru — APPROVED** [review]

File: plone6-classic-nodb/README.md [hunk 1]
```diff
@@ -9,6 +9,26 @@ RelStorage/PostgreSQL is the expected database backend.
 An external database is needed for this setup to work.
 The external database can be located inside or outside of your Kubernetes cluster, or the namespace you use to install this chart, as long as it is accessible.
 
+Database credentials (database name, user name, password) are expected in a secret.
```
Comment (stevepiercy) [inline plone6-classic-nodb/README.md:12]: Is host required as well, or is localhost assumed as default?

```suggestion
Database credentials (database name, user name, and password) are expected in a secret.
```

File: plone6-classic-nodb/README.md [hunk 2]
```diff
@@ -9,6 +9,26 @@ RelStorage/PostgreSQL is the expected database backend.
...
+```
+
+You can override the default secret name in `values.yaml`:
+
+```yaml
+backend:
+  dbsecret:
+    name: <your-secret>
+```
+
```
Comment (stevepiercy) [inline plone6-classic-nodb/README.md:31]:
```suggestion

See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret).

```

File: plone6-classic-nodb/README.md [hunk 3]
```diff
@@ -9,6 +9,26 @@ RelStorage/PostgreSQL is the expected database backend.
 An external database is needed for this setup to work.
 The external database can be located inside or outside of your Kubernetes cluster, or the namespace you use to install this chart, as long as it is accessible.
 
+Database credentials (database name, user name, password) are expected in a secret.
+The secret name defaults to `plonedb`.
+
+[Create the secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret) e.g. using `kubectl`:
```
Comment (stevepiercy) [inline plone6-classic-nodb/README.md:15]:
```suggestion
Using the `kubectl` command, you can create a secret.
```

File: plone6-classic-pg/README.md [hunk 4]
```diff
@@ -13,6 +17,28 @@ postgresql:
     claimName: <your-pvc>
 ```
 
+### Database credentials
+
+Database credentials (database name, user name, password) are expected in a secret.
```
Comment (stevepiercy) [inline plone6-classic-pg/README.md:22]: Is host required as well, or is localhost assumed as default?

```suggestion
Database credentials (database name, user name, and password) are expected in a secret.
```

File: plone6-classic-pg/README.md [hunk 5]
```diff
@@ -13,6 +17,28 @@ postgresql:
...
+```
+
+You can override the default secret name in `values.yaml`:
+
+```yaml
+postgresql:
+  dbsecret:
+    name: <your-secret>
+```
+
```
Comment (stevepiercy) [inline plone6-classic-pg/README.md:41]:
```suggestion

See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret).

```

File: plone6-classic-pg/README.md [hunk 6]
```diff
@@ -13,6 +17,28 @@ postgresql:
     claimName: <your-pvc>
 ```
 
+### Database credentials
+
+Database credentials (database name, user name, password) are expected in a secret.
+The secret name defaults to `plonedb`.
+
+[Create the secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret) e.g. using `kubectl`:
```
Comment (stevepiercy) [inline plone6-classic-pg/README.md:25]:
```suggestion
Using the `kubectl` command, you can create a secret.
```

File: plone6-volto-pg-nginx-varnish/README.md [hunk 7]
```diff
@@ -13,3 +17,24 @@ postgresql:
     claimName: <your-pvc>
 ```
 
+### Database credentials
+
+Database credentials (database name, user name, password) are expected in a secret.
```
Comment (stevepiercy) [inline plone6-volto-pg-nginx-varnish/README.md:22]: Is host required as well, or is localhost assumed as default?

```suggestion
Database credentials (database name, user name, and password) are expected in a secret.
```

File: plone6-volto-pg-nginx-varnish/README.md [hunk 8]
```diff
@@ -13,3 +17,24 @@ postgresql:
...
+The secret name defaults to `plonedb`.
+
+[Create the secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret) e.g. using `kubectl`:
```
Comment (stevepiercy) [inline plone6-volto-pg-nginx-varnish/README.md:25]:
```suggestion
Using the `kubectl` command, you can create a secret.
```

File: plone6-volto-pg-nginx-varnish/README.md [hunk 9]
```diff
@@ -13,3 +17,24 @@ postgresql:
...
+    --from-literal=database-password='plone'
+```
+
+You can override the default secret name in `values.yaml`:
+
+```yaml
+postgresql:
+  dbsecret:
+    name: <your-secret>
+```
```
Comment (stevepiercy) [inline plone6-volto-pg-nginx-varnish/README.md:40]:
````suggestion
```

See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret).

````

File: plone6-volto-pg/README.md [hunk 10]
```diff
@@ -13,6 +17,28 @@ postgresql:
...
+```
+
+You can override the default secret name in `values.yaml`:
+
+```yaml
+postgresql:
+  dbsecret:
+    name: <your-secret>
+```
+
```
Comment (stevepiercy) [inline plone6-volto-pg/README.md:41]:
```suggestion

See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret).

```

File: plone6-volto-pg/README.md [hunk 11]
```diff
@@ -13,6 +17,28 @@ postgresql:
     claimName: <your-pvc>
 ```
 
+### Database credentials
+
+Database credentials (database name, user name, password) are expected in a secret.
+The secret name defaults to `plonedb`.
+
+[Create the secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret) e.g. using `kubectl`:
```
Comment (stevepiercy) [inline plone6-volto-pg/README.md:25]:
```suggestion
Using the `kubectl` command, you can create a secret.
```

File: plone6-volto-pg/README.md [hunk 12]
```diff
@@ -13,6 +17,28 @@ postgresql:
     claimName: <your-pvc>
 ```
 
+### Database credentials
+
+Database credentials (database name, user name, password) are expected in a secret.
```
Comment (stevepiercy) [inline plone6-volto-pg/README.md:22]: Is host required as well, or is localhost assumed as default?

```suggestion
Database credentials (database name, user name, and password) are expected in a secret.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Same three suggestions across the 4 README.mds.

File: plone6-classic-nodb/README.md (hunk 1 above)
Comment (tschorr) [reply plone6-classic-nodb/README.md]: PostgreSQL database cluster host and port are not configured with the secret. I have tried to elaborate on the database configuration in the README. Imo database cluster host and port should not be secret, they are not part of the credentials needed to access the actual database (you can move the database between database clusters without having to change the credentials). Maybe we need to this discuss this later, but let's start like this.

File: plone6-classic-nodb/README.md [hunk 13]
```diff
@@ -6,8 +6,9 @@ RelStorage/PostgreSQL is the expected database backend.
 
 ## Prerequisites
 
-An external database is needed for this setup to work.
+An external PostgreSQL database cluster is needed for this setup to work.
 The external database can be located inside or outside of your Kubernetes cluster, or the namespace you use to install this chart, as long as it is accessible.
+You will need to access to the PostgreSQL database cluster for Plone, see the next section.
```
Comment (stevepiercy) [inline plone6-classic-nodb/README.md:11]:
```suggestion
You will need access to the PostgreSQL database cluster for Plone.
See the next section.
```

File: plone6-classic-nodb/README.md [hunk 14]
```diff
@@ -35,4 +36,28 @@ See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/do
...
 ### Backend
 
+You will need to configure the following options in `values.yaml` or on the command line:
+
+```yaml
+backend:
+  databasehost: your.db.host
+```
+The PostgreSQL database cluster host.
+There is no sensible default for this option, you will have to modify it.
```
Comment (stevepiercy) [inline plone6-classic-nodb/README.md:46]:
```suggestion
There is no sensible default for this option.
You will have to modify it.
```

File: plone6-classic-nodb/README.md (hunk 1 above)
Comment (stevepiercy) [reply plone6-classic-nodb/README.md]: I don't have an opinion. I wanted to make sure that the omission was deliberate, not an oversight, and that the host was defined somewhere. I don't know how to connect to a database without a host parameter.

Should the host configuration be applied to the other three README.mds?

File: plone6-classic-nodb/README.md (hunk 13 above)
Comment (tschorr) [reply plone6-classic-nodb/README.md:11]: argh there's a missing 'configure' in my sentence, it should be: 'You will need to configure access to the ...'. Does that have an impact on your suggestion?

File: plone6-classic-nodb/README.md (hunk 1 above)
Comment (tschorr) [reply plone6-classic-nodb/README.md]: > Should the host configuration be applied to the other three README.mds?

No, at least not in the same fashion. The other charts come with an included database, so there is no need to configure host + port.

File: plone6-classic-nodb/README.md (hunk 13 above)
Comment (stevepiercy) [reply plone6-classic-nodb/README.md:11]: Updated suggestion.

**Review by stevepiercy — APPROVED** [review]
Thank you!

**Pull request merged**